### DISCUSSION TITLE
Query on optimising performance for client components

### DISCUSSION THREAD

**Discussion by lmac-1** [amannn/next-intl#859] [Q&A]
In our application, we have some fairly interactive parts of the site which are mainly client components. I just wanted to clarify the performance impact and whether we are doing things in the "right" way. 

### What we've built so far
We have a fairly interactive part of the application (`/feedback` with various subroutes). The pages within these routes are client components due to having a lot of interactivity. 

We are wrapping the `layout.tsx` file on this route with the `NextIntlClientProvider` and picking the feedback messages to pass through: 
```
import pick from 'lodash.pick';
import { NextIntlClientProvider, useMessages } from 'next-intl';

export default function FeedbackLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  const messages = useMessages();
  return (
    <NextIntlClientProvider messages={pick(messages, 'feedback')}>
      <div className="mx-auto my-10 w-full max-w-md px-3 sm:px-0">
        {children}
      </div>
    </NextIntlClientProvider>
  );
}

```

### My question
Let's say on a page component I am calling `useTranslations`: 
`const t = useTranslations('feedback');`

The `feedback` strings contain all the strings needed for that page. Given this context, are there performance implications if I call `useTranslations` again in child components? Or should I be passing down only the specific strings needed for each child component from the parent page component (for example in a `translations` prop which holds an object of strings needed in that child component)?

**Comment by amannn** [comment]
Sounds absolutely fine to me what you’ve built so far! `useTranslations` is designed to be called in every component where you need internationalization.

Did you encounter any performance issues since you’re asking?
